### DISCUSSION TITLE
What is the function of  .browser_modules?

### DISCUSSION THREAD

**Discussion by zhaomenghuan** [eclipse-theia/theia#8697] [Q&A] [answered]
The project under the Electron target will generate a .browser_modules folder, but the specific reason has not been found. Can someone tell me what generated this folder? What is the function of  .browser_modules?

**Comment by vince-fugnitto** [comment, marked as answer]
I believe the `.browser_modules` contain the native node modules for web in this instance (when building the application).

When we run `yarn rebuild:electron` or `yarn rebuild:browser` these are the modules that are essentially rebuilt, so we can change targets between `electron` and `browser` targets.

At this time, they contain the following:

- `'@theia/node-pty'`
- ` 'nsfw'`
- `'native-keymap'`
- `'find-git-repositories'`
- `'drivelist'`
